src/opm/input/eclipse/EclipseState/Grid/SatfuncPropertyInitializers.cpp

```diff
@@ -395,6 +426,19 @@ namespace {
                                       table.getKrwColumn(), tolcrit);
     }
 
+    /// Maximum water saturation for which Krgw(1-Sw) <= tolcrit.
+    ///
+    /// Expected Table Format:
+    ///    [1-Sw,  Krgw(1-Sw), ...other...]
+    ///
+    ///    Krgw decreasing.
+    template <typename T>
+    double critical_water_sgwfn(const T& table, const double tolcrit)
+    {
+        return crit_sat_decreasing_KR(table.getSgColumn(),
+                                      table.getKrgwColumn(), tolcrit);
```

Comment:
This will return `Sg` at the critical relative permeability level.  You'll have to transform that into a water saturation (`1 - Sg`) at some point.